toqito/nonlocal_games/binary_constraint_system_game.py

```diff
@@ -0,0 +1,177 @@
+    ``M`` of shape (m, n) and the corresponding element of the binary vector ``b``
+    into an n-dimensional tensor of shape (2, 2, ..., 2) (one axis per variable).
+
+    The conversion works as follows:
+      1. For the i-th constraint, compute the constant value as ``rhs = (-1)**(b[i])``.
+      2. Create an n-dimensional array (tensor) of shape ``(2,)*n`` filled with ``-rhs``.
+      3. Compute the index from the first n entries of the i-th row of ``M`` by taking each value modulo 2.
+      4. Set the tensor element at that index to ``rhs``.
+
+    For example:
+      If ``M[i] = [1, 1]`` and ``b[i] = 0`` (so ``rhs = 1``):
+      - The tensor is of shape (2, 2) and is created as:
+            [ [-1, -1],
+              [-1, -1] ]
+      - The index is computed as ``(1 % 2, 1 % 2) = (1, 1)``.
+      - At position (1, 1), the value is set to 1, resulting in:
+            [ [-1, -1],
+              [-1,  1] ]
+
+    This tensor now represents the constraint in full detail.
+
+    Examples
+    ==========
+    .. jupyter-execute::
+    
+     import numpy as np
+     from toqito.nonlocal_games.binary_constraint_system_game import create_bcs_constraints
+          
+     M = np.array([[1, 1], [1, 1]], dtype=int)
+     b = np.array([0, 1], dtype=int)
+     constraints = create_bcs_constraints(M, b)
+     constraints[0].shape
+          
+    :param M: A binary matrix of shape (m, n) defining which variables appear in each constraint.
+    :param b: A binary vector of length m that determines the constant term ``(-1)**(b[i])``.
+    :return: A list of NumPy arrays, each of shape ``(2,)*n``. Each tensor represents
+             one constraint in tensor form.
+             
+    """
+    m, n = M.shape
+    constraints = []
+    for i in range(m):
+        rhs = (-1) ** b[i]
+        constraint_tensor = np.full((2,) * n, fill_value=-rhs, dtype=int)
+        idx = tuple(M[i] % 2)
+        constraint_tensor[idx] = rhs
+        constraints.append(constraint_tensor)
+    return constraints
+
+
+def generate_solution_group(M: np.ndarray, b: np.ndarray) -> tuple[list[int], list[int]]:
+    r"""Generate a bitmask representation for a binary constraint system (BCS) game.
+
+    This function converts each row of the binary matrix ``M`` into an integer bitmask,
+    pairing it with the corresponding parity from ``b``. The bitmask representation
+    can be useful for analyzing linear system games.
+
+    Examples
+    ========
+    .. jupyter-execute::
+    
+     import numpy as np
+     from toqito.nonlocal_games.binary_constraint_system_game import create_bcs_constraints
+
+     M = np.array([[1, 1, 0], [0, 1, 1]])
+     b = np.array([0, 1])
+     row_masks, parity = generate_solution_group(M, b)
+
+     print("Row masks:", row_masks)
+     print("Parity:", parity)
+       
+     The method used to determine the existence of a perfect commuting strategy was originally introduced in :cite:`Cleve_2016_Perfect`.
```

Comment:
RTD will still fail because this is indented at the same level as what is within the `..jupyter-execute::` block. Also, please refer to the other `jupyter-execute` blocks within `toqito` to see how this is formatted elsewhere as it seems like the build keeps failing for similar reasons.